### DISCUSSION TITLE
Override controller methods

### DISCUSSION THREAD

**Discussion by PabloRuiz-TT** [Azure/azure-mobile-apps#922] [Q&A] [answered]
Hello community, could someone tell me if in the documentation there is a way to override the existing controller methods when working with : TableController

**Comment by adrianhall** [comment]
All the controller methods are virtual.

I would like to understand the use case that would cause this, however.  If it is a common use case, I may just add a simplification for it.

**Reply by PabloRuiz-TT** [reply]
I am looking to overwrite these methods to return from the default get method, a viewmodel or at the time of making a post, receive a dto (currently my post model of one of my tables in swagger forces me to fill fields of its classes with which it has a relationship).

**Reply by adrianhall** [reply, marked as answer]
In that case, I think you would be better off using a wrapper around the EntityTableRepository<T>.  Take a look at the MappedRepository (which is based on Automapper) as an example.  That one is only a couple of dozen lines of code and does what you want.  Even if you don't want to use Automapper, you can use that as a guide to implementing your own transforms.
